clients/src/test/java/org/apache/kafka/clients/MockClient.java

```diff
@@ -461,6 +462,7 @@ public ClientRequest newClientRequest(String nodeId, AbstractRequest.Builder<?>
     @Override
     public ClientRequest newClientRequest(String nodeId, AbstractRequest.Builder<?> requestBuilder, long createdTimeMs,
                                           boolean expectResponse, RequestCompletionHandler callback) {
+        requestCount.incrementAndGet();
```

Comment:
When do we decrement this?

Reply:
Does it need a way to decrement the count of a class used for testing?

Reply:
I may have misunderstood the intent. I thought it was meant to count in-flight requests, but seems it is a total count instead?

Reply:
bq. seems it is a total count instead?
yep. used only for testing so it should be fine without decrement call

Reply:
I think a total count is fine.  Maybe name it `totalRequestCount` to make it obvious that that is what it is.

As @hachikuji commented, there is really no need to share a `MockAdminClient` across multiple tests.  The mock client is extremely lightweight-- it doesn't manage any threads or sockets.

Reply:
Yes, `totalRequestCount` sounds clearer. We might also consider setting it back to 0 in `MockClient.reset`.

Reply:
Thanks for all reviews. Will address the following comments
1) requestCount -> totalRequestCount
2) set totalRequestCount to zero in MockClient.reset